client/src/main/java/org/apache/iotdb/client/AbstractClient.java

```diff
@@ -478,15 +478,17 @@ private static void printBlockLine(boolean printTimestamp, int colCount,
     } else {
       blockLine.append("+");
     }
-    if (resultSetMetaData.getColumnName(2).equals(GROUPBY_DEVICE_COLUMN_NAME)) {
-      maxValueLength = measurementColumnLength;
-    } else {
-      int tmp = Integer.MIN_VALUE;
-      for (int i = 1; i <= colCount; i++) {
-        int len = resultSetMetaData.getColumnLabel(i).length();
-        tmp = Math.max(tmp, len);
+    if(printTimestamp) {
+      if (resultSetMetaData.getColumnName(2).equals(GROUPBY_DEVICE_COLUMN_NAME)) {
+        maxValueLength = measurementColumnLength;
+      } else {
+        int tmp = Integer.MIN_VALUE;
+        for (int i = 1; i <= colCount; i++) {
+          int len = resultSetMetaData.getColumnLabel(i).length();
+          tmp = Math.max(tmp, len);
+        }
+        maxValueLength = tmp;
       }
-      maxValueLength = tmp;
     }
     if(printTimestamp) {
```

Comment:
Why not merge this if(printTimestamp) into next?

Reply:
And I try your version, it seems that you didn't fix the "show storage group" problem. It still show the error message.

Reply:
> And I try your version, it seems that you didn't fix the "show storage group" problem. It still show the error message.

The problem has been fixed. Perhaps you need to recompile the project.

Reply:
Sorry for that, I try it again and it really works.
And you can split the `printBlockLine` function into two functions.
Because, the logic in that function is completely split by the `printTimestamp`.
<img width="821" alt="image" src="https://user-images.githubusercontent.com/16079446/72213191-538c7f00-3525-11ea-93a3-0f82a8d883cd.png">

Reply:
The "create timeseries" statement with the "compression" argument can't still print successfully.

Reply:
it can create successfully, but it does the wrong thing, I think we should choose one bewteen syntax `compression` and `compressor`.
<img width="1103" alt="image" src="https://user-images.githubusercontent.com/16079446/72213235-dca3b600-3525-11ea-908e-3f8096ef1b68.png">

Reply:
Compression is a wrong syntax, the wrong syntax doesn't mean it gives an error message. Instead, It is identified as a normal property and doesn't influence anything. So if you use `create timeseries root.sg1.d2.s1 with datatype=INT64, encoding=RLE, compression=SNAPPY;`, root.sg1.d2.s1 will use UNCOMPRESSED.